File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }
        std::unordered_map<char, int> freqTable;

        for(const auto& c : s){
            freqTable[c]++;
        }

        for(const auto& c : t){
            freqTable[c]--;
        }
        for(const auto& value : freqTable){
            if(value.second != 0){
                return false;
            }
            return true;
        }

    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freqTable = {}
        for num in nums:
            freqTable[num] = freqTable.get(num, 0) + 1
        n = len(nums) + 1
        buckets = [[] for _ in range(n)]
        for key, val in freqTable.items():
            buckets[val].append(key)
        buckets.reverse()
        result = []
        for bucket in buckets:
            for element in bucket:
                if len(result) < k:
                    result.append(element)
                else:
                    return result
        return result
            